SnappyAPITests/CameraManagerTests.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  CameraManagerTests.swift
+//  Snappy
+//
+//  Created by Pawel Chmiel on 18.10.2016.
+//  Copyright © 2016 Droids On Roids. All rights reserved.
+//
+
+import XCTest
+import AVFoundation
+@testable import Snappy
+
+class CameraManagerTests: XCTestCase {
+    
+    var previewSize = CGSize(width: 200, height: 300)
+    override func setUp() {
+        super.setUp()
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+    
+    func testScaleImage() {
+        let image = UIImage().scaledToSize(previewSize)
+        XCTAssertTrue(image.size.width == 200 && image.size.height == 300)
+    }
+    
+    func testSetCamera() {
+        CameraManager.generateCameraPreview(previewSize: previewSize) { view in
+            XCTAssertTrue(view == CameraManager.previewView)
+        }
+    }
+    
+    func testWorkingOutput() {
+        CameraManager.generateCameraPreview(previewSize: previewSize) { view in
+            XCTAssertNotNil(view)
+        }
+    }
+
+    func testZTakePhoto() {
+        let exp = expectation(description: "take photo")
+        CameraManager.takePhoto { image in
+            if let image = image {
+                XCTAssertTrue(image.size.width > 1000)
+            }
+            exp.fulfill()
+        }
+        
+        waitForExpectations(timeout: 5.0, handler: nil)
+    }
+    
+    func testAToggleFlashMode() {
+        if let device = (CameraManager.session!.inputs.first as! AVCaptureDeviceInput).device {
```

Comment:
let's use `?` instead of `!` in `if let` + this test is only executed when 2 checks passes, it has to assert even tho the checks don't pass